### PR TITLE
DNSPolicy API changes

### PR DESCRIPTION
**Pull request by maleck13**
RFC for some identified DNSPolicy changes prior to v1

### CODE REVIEW

File: rfcs/dns-policy-api-changes.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# RFC DNSPolicy v1 changes and improvements
+
+- Feature Name: (fill me in with a unique ident, `dns_policy_api_changes`)
```
Comment (mikenairn) [inline rfcs/dns-policy-api-changes.md:3]:
```suggestion
- Feature Name: (`dns_policy_api_changes`)
```

File: rfcs/dns-policy-api-changes.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+- Start Date: (fill me in with today's date, 2024-09-11)
```
Comment (mikenairn) [inline rfcs/dns-policy-api-changes.md:4]:
```suggestion
- Start Date: (2024-09-11)
```

**Comment by mikenairn** [comment]
This looks fine and makes sense to me :+1:

**Review by mikenairn — APPROVED** [review]

File: rfcs/dns-policy-api-changes.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    geo: US # geo for listeners targeted 
+    defaultGEO: true # should this be consisdered the default GEO for the listener hosts
+  providerRefs:
+    - name: aws-credential-secret    
+```
+
+So no longer do you need to specify whether the policy is a load balanced one or not via the redundant `routingStrategy` field
+
+
+Now you simplify specify the weight you want to use for the listeners in the gateway, the geo you want to use and whether it should be used as a default GEO or not (this is used by some cloud providers as a catch-all option if a user from a none specified GEO does a DNS lookup.). Each of the fields under "loadbalancing" will now be required.
```
[the remark this answers is not in the document]
Comment (mikenairn) [reply rfcs/dns-policy-api-changes.md:106]: The behaviour is that we require each Gateway/DNSPolicy to tell us both the weight and geo that is to be used by any load balanced records created for it, we don't/can't guess it. Currently this is done through a combination of attributes in the policy (defaultWeight and defaultGeo) and labels on the Gateway resource itself. The proposal here is only changing where that data comes from, which will always be the policy now.

Maybe it's a confusion over what `defaultGEO` means? This is a flag to say whether or not the current geo (which you have to set) is what the wildcard global endpoint should be pointing to. This internally means whether or not this endpoint should be included in the endpoint list of DNSRecords created for this policy. i.e. geo = "ie", defaultGeo=true, add this:
```
- dnsName: klb.loadbalanced.mn.hcpapps.net
  providerSpecific:
    - name: geo-code
      value: '*'
  recordTTL: 300
  recordType: CNAME
  setIdentifier: default
  targets:
    - ie.klb.loadbalanced.mn.hcpapps.net
``` 

For providers that support it (Route53, Azure), this sets up the recordset to have a default wildcard endpoint where any traffic coming from regions that do not have a specific geo are directed.

[the remark this answers is not in the document]
Comment (azgabur) [reply rfcs/dns-policy-api-changes.md:106]: Now it makes sense. I was confused about the `defaultGEO` meaning. In that case I now understand that both `weight` and `geo` code needs to be set. Thanks for explanation :+1:

[the remark this answers is not in the document]
Comment (maleck13) [reply rfcs/dns-policy-api-changes.md:106]: DNSPolicies are not related. They target different gateways. Likely on different clusters

**Review by azgabur — APPROVED** [review]

**Pull request merged**